src/passes/OptimizeInstructions.cpp

```diff
@@ -1976,6 +1977,12 @@ struct OptimizeInstructions
              op == Abstract::getBinary(type, Abstract::LtS) ||
              op == Abstract::getBinary(type, Abstract::GeS) ||
              op == Abstract::getBinary(type, Abstract::GtS))) {
+          // if (auto* c = inner->right->dynCast<Const>()) {
+          //   // skip signed(x - min_s) <=> 0 case
+          //   if (c->value.isSignedMin()) {
+          //     return nullptr;
+          //   }
+          // }
```

Comment:
I don't think this is the right approach. This only works if we know the constant at compile time. But we need to handle the case of a value only known at runtime.

Let me open a PR with what I think we have to do.

Reply:
Yes, that's already outdated approach. With recent commit I removed simplification for `x - y <> 0` cases. And only leave for `x - y == 0` and `x - y != 0`